src/CSharp/CodeCracker/Design/InconsistentAccessibility/InconsistentAccessibilityInMethodReturnType.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Threading.Tasks;
+using CodeCracker.Properties;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace CodeCracker.CSharp.Design.InconsistentAccessibility
+{
+    public sealed class InconsistentAccessibilityInMethodReturnType : InconsistentAccessibilityInfoProvider
+    {
+        private static readonly LocalizableString CodeActionMessage = new LocalizableResourceString(nameof(Resources.InconsistentAccessibilityInMethodReturnType_CodeActionMessage), Resources.ResourceManager, typeof(Resources));
+
+        public async Task<InconsistentAccessibilityInfo> GetInconsistentAccessibilityInfoAsync(Document document, Diagnostic diagnostic, CancellationToken cancellationToken)
+        {
+            var result = new InconsistentAccessibilityInfo();
+            var syntaxRoot = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var methodThatRaisedError = syntaxRoot.FindNode(diagnostic.Location.SourceSpan).DescendantNodesAndSelf().OfType<MethodDeclarationSyntax>().FirstOrDefault();
+            if (methodThatRaisedError != null)
+            {
+                result.TypeToChangeAccessibility = methodThatRaisedError.ReturnType;
+                result.CodeActionMessage = string.Format(CodeActionMessage.ToString(), methodThatRaisedError.ReturnType, methodThatRaisedError.GetIdentifier().ValueText);
+                result.NewAccessibilityModifiers = methodThatRaisedError.CloneAccessibilityModifiers();
+            }
+
```

Comment:
You will have to verify if the return type is in source, the same problem from `CC0051`.

MyDependence.dll:

````csharp
    public class DependedUpon
    {
        protected class A { }
    }
````

My code:

````csharp
    public class Class1 : ClassLibrary6.DependedUpon
    {
        public void Foo()
        {
            var a = new A();
        }
        public A Bar()
        {
            throw new Exception();
        }
    }
````

I can't change the accessibility of `DependedUpon.A` to be public, it is not in my source.